### PR TITLE
Spec and Impl Fixes for Event Catcher

### PR DESCRIPTION
**Pull request by jshaughn**
- move to use collector in spec
- make @collector a field on Stream

@israel-hdez @josejulio @lucasponce Note, there are a few impl changes along with the spec fixes.

### CODE REVIEW

File: spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/stream_spec.rb [hunk 1]
```diff
@@ -1,151 +1,112 @@
...
-      :options     => { :tenant => 'hawkular' }
-    )
-    allow(client).to receive(:metrics).and_return(stubbed_metrics_client)
-    allow(client).to receive(:inventory).and_return(stubbed_inventory_client)
-    client
+  let(:collector) do
+    ManageIQ::Providers::Hawkular::Inventory::Collector::MiddlewareManager
+      .new(ems_hawkular, ems_hawkular)
+      .tap do |collector|
+        allow(collector).to receive(:resources_for) do |arg|
```
Comment (josejulio) [inline spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/stream_spec.rb:89]: That's a nice way of mocking a method

**Comment by lucasponce** [comment]
I am testing locally and I found a problem in some reports code.
Not sure if related or not, but as yesterday worked, let me investigate first.

**Comment by lucasponce** [comment]
No, it is OK.
LGTM

**Comment by collector** [comment]
Please note to remove me from your issue. I was wrongly marked and get
these emails here.

2017-12-13 15:14 GMT+02:00 Lucas Ponce <notifications@github.com>:

**Pull request merged**